LabKey/labkey-ui-components
# Issue 41718: List Domain Designer Field imports should observe auto-increment fields

#### Rationale
See [issue](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=41718) for more details.
Manual test steps:
- Create a new list, and choose an auto integer key. Export the fields as one.fields.json
- Click to create the above new list. Navigate back to the design. Export the fields as two.fields.json
- When creating a new list, importing either one.fields.json or two.fields.json will set the key field to Data Type 'Auto Increment'
- Newly created list from either one.fields.json or two.fields.json does in fact have an auto increment key
- Note that existing lists with an auto integer key will still export a key field of Data Type 'Integer'. Only new lists will export 'Auto Increment' keys appropriately. Due to the small scope of this issue however this limitation is acceptable.

#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-components/pull/1370
- https://github.com/LabKey/labkey-ui-premium/pull/280
- https://github.com/LabKey/platform/pull/5051
- https://github.com/LabKey/biologics/pull/2583
- https://github.com/LabKey/sampleManagement/pull/2303
- https://github.com/LabKey/inventory/pull/1130

#### Changes
- Add AUTO_INT_CONCEPT_URI const
- Ensure AUTO_INT_CONCEPT_URI is set as conceptURI for new PK fields
- Ensure key's conceptURI for .field.json import is checked for AUTO_INT_CONCEPT_URI value, at which point we set dataType to AUTOINT_TYPE
- Bump version
- components.md
